x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -88,6 +88,17 @@ func NewMongoCrypt(opts *options.MongoCryptOptions) (*MongoCrypt, error) {
 		C.mongocrypt_setopt_bypass_query_analysis(wrapped)
 	}
 
+	var keyExpirationMs uint64 = 60_000 // 60,000 ms
+	if opts.KeyExpiration != nil {
+		expirationMs := opts.KeyExpiration.Milliseconds()
+		if expirationMs < 0 {
+			keyExpirationMs = 0
+		} else {
+			keyExpirationMs = uint64(expirationMs)
+		}
+	}
+	C.mongocrypt_setopt_key_expiration(crypt.wrapped, C.uint64_t(keyExpirationMs))
```

Comment:
Should we expose this type as a uint64 instead of a time.Duration in the options API? That is the convention for both C and Rust:

Rust: https://github.com/mongodb/libmongocrypt-rust/pull/38/files
C: https://github.com/mongodb/mongo-c-driver/commit/b0edf300b8623bc8a4d861f02aaefc8c312c46c9#diff-782e2ab0e9ed19e1fc3e8e7bb5587e57805d5c816b85ab4b606da41de292bc77R137

Reply:
IMO, `time.Duration` is more expressive. We also use it for duration values in other option structs. However, I'm open to using uint64 if it simplifies the API and improves consistency with other drivers.

Reply:
It's conventional in the Go Driver to use the `time.Duration` type to specify duration values (e.g. [FindOptions.SetMaxTime](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo/options#FindOptions.SetMaxTime)). There is a difference in precision (nanoseconds vs milliseconds), but it's unlikely any users want to set sub-millisecond key expiration values. Using `time.Duration` avoids the more likely arithmetic bugs that can happen when manually converting minutes, hours, or days to milliseconds (e.g. you get to use `2 * time.Hour` vs `2 * 60 * 60 * 1_000`).

For those reasons, I think we should use `time.Duration`.

Reply:
We have latitude to modify the spec's suggested API to keep it idiomatic for Go. No [option](https://pkg.go.dev/go.mongodb.org/mongo-driver/v2/mongo/options) in the current Go Driver API accepts a value in milliseconds; all use `time.Duration`.

Rust actually uses a `Duration` value for the DEK cache lifetime API in their driver. See the code [here](https://github.com/mongodb/mongo-rust-driver/pull/1284/files#diff-0479cfbb266e216249d562a09dd79bc152029307663b13ec6cf7d5b4cd70987bR108). The above linked code in `github.com/mongodb/libmongocrypt-rust` appears to be the Rust wrapper library for libmongocrypt.